lightning/src/ln/max_payment_path_len_tests.rs

```diff
@@ -0,0 +1,312 @@
+	check_added_monitors!(nodes[0], 1);
+	let mut events = nodes[0].node.get_and_clear_pending_msg_events();
+	assert_eq!(events.len(), 1);
+	let path = &[&nodes[1], &nodes[2]];
+	let args = PassAlongPathArgs::new(&nodes[0], path, amt_msat, payment_hash_2, events.pop().unwrap())
+		.with_payment_secret(payment_secret_2)
+		.with_payment_metadata(recipient_onion_allws_2_hops.payment_metadata.unwrap());
+	do_pass_along_path(args);
+	claim_payment_along_route(&nodes[0], &[&[&nodes[1], &nodes[2]]], false, payment_preimage_2);
+}
+
+#[test]
+fn one_hop_blinded_path_with_custom_tlv() {
+	// Test that we'll limit our maximum path length when paying to a 1-hop blinded path based on the
+	// size of the provided custom TLV, and refuse to send at all prior to pathfinding if it's too
+	// large.
+	let chanmon_cfgs = create_chanmon_cfgs(3);
+	let node_cfgs = create_node_cfgs(3, &chanmon_cfgs);
+	let node_chanmgrs = create_node_chanmgrs(3, &node_cfgs, &[None, None, None]);
+	let nodes = create_network(3, &node_cfgs, &node_chanmgrs);
+	create_announced_chan_between_nodes(&nodes, 0, 1);
+	let chan_upd_1_2 = create_announced_chan_between_nodes_with_value(&nodes, 1, 2, 1_000_000, 0).0.contents;
+
+	// Construct the route parameters for sending to nodes[2]'s 1-hop blinded path.
+	let amt_msat = 100_000;
+	let (payment_preimage, payment_hash, payment_secret) = get_payment_preimage_hash(&nodes[2], Some(amt_msat), None);
+	let payee_tlvs = ReceiveTlvs {
+		payment_secret,
+		payment_constraints: PaymentConstraints {
+			max_cltv_expiry: u32::max_value(),
+			htlc_minimum_msat: chan_upd_1_2.htlc_minimum_msat,
+		},
+		payment_context: PaymentContext::unknown(),
+	};
+	let mut secp_ctx = Secp256k1::new();
+	let blinded_path = BlindedPath::one_hop_for_payment(
+		nodes[2].node.get_our_node_id(), payee_tlvs, TEST_FINAL_CLTV as u16,
+		&chanmon_cfgs[2].keys_manager, &secp_ctx
+	).unwrap();
+	let route_params = RouteParameters::from_payment_params_and_value(
+		PaymentParameters::blinded(vec![blinded_path.clone()]),
+		amt_msat,
+	);
+
+	// Calculate the maximum custom TLV value size where a valid onion packet is still possible.
+	const CUSTOM_TLV_TYPE: u64 = 65537;
+	let final_payload_len_without_custom_tlv = msgs::OutboundOnionPayload::BlindedReceive {
+		sender_intended_htlc_amt_msat: amt_msat,
+		total_msat: amt_msat,
+		cltv_expiry_height: nodes[0].best_block_info().1 + TEST_FINAL_CLTV,
+		encrypted_tlvs: &blinded_path.1.blinded_hops[0].encrypted_payload,
+		intro_node_blinding_point: Some(blinded_path.1.blinding_point),
+		keysend_preimage: None,
+		custom_tlvs: &Vec::new()
+	}.serialized_length();
+	let max_custom_tlv_len = 1300
+		- crate::util::ser::BigSize(CUSTOM_TLV_TYPE).serialized_length() // custom TLV type
+		- crate::util::ser::BigSize(1200).serialized_length() // custom TLV length
+		- 2 // onion payload varint prefix increased ser size due to custom TLV
+		- 1 // ?? why is this needed
```

Comment:
FYI - still not sure why I'm having to subtract an extra byte here, hopefully will push an update soon.